tensorflow_addons/layers/polynomial.py

```diff
@@ -50,8 +52,14 @@ class PolynomialCrossing(tf.keras.layers.Layer):
         ```
 
     Arguments:
-        projection_dim: project dimension. Default is `None` such that a full
-          (`input_dim` by `input_dim`) matrix is used.
+        projection_dim: project dimension to reduce the computational cost.
+          Default is `None` such that a full (`input_dim` by `input_dim`)
+          matrix W is used. If enabled, a low-rank matrix W = U*V will be used,
+          where U is of size `input_dim` by `projection_dim` and V is of size
+          `projection_dim` by `input_dim`. `projection_dim` need to be smaller
+          than `input_dim`/2 to improve the model efficiency.
```

Comment:
Not sure if I follow, `projection_dim` semantics and default value is not changed, we merely add the capacity for lower-rank decomposition?

Reply:
Yeah I meant is there a benefit for that to be configurable or could we set an optimal decomposition? In either case not worth the braking API change it would cause.